src/Helpers/JWKFetcher.php

```diff
@@ -59,101 +62,135 @@ protected function convertCertToPem($cert)
+                'method' => 'GET',
+                'guzzleOptions' => $this->guzzleOptions
+            ]);
+            $jwks    = $request->call();
+            foreach ($jwks['keys'] as $key) {
+                $secret[$key['kid']] = $this->convertCertToPem($key['x5c'][0]);
+            }
 
-        // Need the kid and x5c for a well-formed return value. 
-        if (! is_null($kid) && ! is_null($x5c)) {
-            $secret[$kid] = $this->convertCertToPem($x5c);
-            $this->cache->set($jwks_url, $secret);
+            $this->cache->set($url, $secret);
         }
 
         return $secret;
     }
+    // phpcs:enable
 
     /**
-     * Get a specific property from a JWKS using a key, if provided.
+     * Fetch x509 cert for RS256 token decoding.
      *
-     * @param array       $jwks JWKS to parse.
-     * @param string      $prop Property to retrieve.
-     * @param null|string $kid  Kid to check.
+     * @param string      $jwks_url URL to the JWKS.
+     * @param string|null $kid      Key ID.
      *
-     * @return null|string
+     * @return string|null
      */
-    public function getProp(array $jwks, $prop, $kid = null)
+    public function getJwksX5c($jwks_url, $kid = null)
     {
-        $r_key = null;
-        if (! $kid) {
-            // No kid indicated, get the first entry.
-            $r_key = $jwks['keys'][0];
-        } else {
-            // Iterate through the JWKS for the correct kid.
-            foreach ($jwks['keys'] as $key) {
-                if (isset($key['kid']) && $key['kid'] === $kid) {
-                    $r_key = $key;
-                    break;
-                }
-            }
+        $cache_key = $jwks_url.'|'.(string) $kid;
+
+        $x5c = $this->cache->get($cache_key);
+        if (! is_null($x5c)) {
+            return $x5c;
         }
 
-        // If a key was not found or the property does not exist, return.
-        if (is_null($r_key) || ! isset($r_key[$prop])) {
+        $jwks = $this->getJwks($jwks_url);
+        $jwk  = $this->getJwk($jwks, $kid);
+
+        if (! $this->subArrayHasOneElement($jwk, 'x5c')) {
             return null;
         }
 
-        // If the value is an array, get the first element.
-        return is_array( $r_key[$prop] ) ? $r_key[$prop][0] : $r_key[$prop];
+        $x5c = $this->convertCertToPem($jwk['x5c'][0]);
+        $this->cache->set($cache_key, $x5c);
+        return $x5c;
     }
 
     /**
      * Get a JWKS given a domain and path to call.
      *
-     * @param string $domain Base domain for the JWKS, including scheme.
-     * @param string $path   Path to the JWKS from the $domain.
+     * @param string $jwks_url URL to the JWKS.
      *
      * @return mixed|string
+     *
+     * @throws RequestException If $jwks_url is empty or malformed.
+     * @throws ClientException  If the JWKS cannot be retrieved.
+     *
+     * @codeCoverageIgnore
      */
-    protected function getJwks($domain, $path)
+    protected function getJwks($jwks_url)
```

Comment:
suggestion: I'd call this "fetch" or "request" jwks. Since it's performing an external query

Reply:
So does `getJwkX5c` ... change that too?

Reply:
If you feel like it, go ahead. Remember that's the method you want to expose to the user (the public method). These other in turn are internal (protected methods)